### PR TITLE
rh_subscriprion bootstrap script

### LINKED ISSUE #1261: Add the rh_subscription cloud-init module to the bootstrap script
In order to have RHEL machines working jointly with OSM, we need to provide the rh_subscription module to the bootstrap script.

### PR DESCRIPTION
**Pull request by mlavacca**
**What this PR does / why we need it**:
Addition of the rh_subscription cloud-init module to the yum
version of the bootstrap script.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Fixes #1261 

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note
RHEL in Azure has been updated to the 8.5 version, and the rh_subscription module has been added to the bootstrap cloud-init when using RHEL.
```

### CODE REVIEW

File: pkg/controller/machine/bootstrap.go [hunk 1]
```diff
@@ -238,7 +247,15 @@ fi
 {{- if .EnterpriseLinux }}
 yum install epel-release -y
 {{- end }}
+
+{{- if and (eq .ProviderSpec.CloudProvider "azure") (eq .ProviderSpec.OperatingSystem "rhel") }}
+yum --disablerepo='*' remove -y 'rhui-azure-rhel8'
+wget https://rhelimage.blob.core.windows.net/repositories/rhui-microsoft-azure-rhel8.config
+yum --config=rhui-microsoft-azure-rhel8.config install -y rhui-azure-rhel8
+{{- end }}
```
Comment (embik) [inline pkg/controller/machine/bootstrap.go:255]: This seems to enable Extended Update Support (EUS) as per https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/redhat/redhat-rhui, which seems to be only necessary if you want to pin to a specific minor release. Why are we adding it?

Comment (embik) [reply pkg/controller/machine/bootstrap.go:255]: Plus, will this even work on images that do "bring your own license"?

Comment (mlavacca) [reply pkg/controller/machine/bootstrap.go:255]: Actually it is not related to the EUS, I used a different config from the one mentioned in the post (it misses the -eus suffix). It was needed to make the whole thing work as yum wasn't able to fetch metadata packages. This is the only way that I found, removing the repo and adding it again

Comment (embik) [reply pkg/controller/machine/bootstrap.go:255]: Ah, fair point. Did not notice that. I'm somewhat confused by that though.

Any idea if we support https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/redhat/byos? I'm not very familiar with RHEL on Azure. Maybe we can improve this snippet by only running it when yum has that specific repo configured?

Comment (moadqassem) [reply pkg/controller/machine/bootstrap.go:255]: @mlavacca what was the error that you were getting ?

Comment (mlavacca) [reply pkg/controller/machine/bootstrap.go:255]: @moadqassem 
```
sudo yum install curl
Updating Subscription Management repositories.
Red Hat Enterprise Linux 8 for x86_64 - BaseOS from RHUI (RPMs)                  0.0  B/s |   0  B     00:00     Errors during downloading metadata for repository 'rhui-rhel-8-for-x86_64-baseos-rhui-rpms':
- Curl error (56): Failure when receiving data from the peer for https://rhui-3.microsoft.com/pulp/repos/content/dist/rhel8/rhui/8/x86_64/baseos/os/repodata/repomd.xml [OpenSSL SSL_read: error:14094415:SSL routines:ssl3_read_bytes:sslv3 alert certificate expired, errno 0]
- Curl error (56): Failure when receiving data from the peer for https://rhui-1.microsoft.com/pulp/repos/content/dist/rhel8/rhui/8/x86_64/baseos/os/repodata/repomd.xml [OpenSSL SSL_read: error:14094415:SSL routines:ssl3_read_bytes:sslv3 alert certificate expired, errno 0]
- Curl error (56): Failure when receiving data from the peer for https://rhui-2.microsoft.com/pulp/repos/content/dist/rhel8/rhui/8/x86_64/baseos/os/repodata/repomd.xml [OpenSSL SSL_read: error:14094415:SSL routines:ssl3_read_bytes:sslv3 alert certificate expired, errno 0]
- Error: Failed to download metadata for repo 'rhui-rhel-8-for-x86_64-baseos-rhui-rpms': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
```

Comment (embik) [reply pkg/controller/machine/bootstrap.go:255]: Is it possible that the problem is that the default RHEL image on Azure is BYOS, not RHUI?

https://github.com/kubermatic/machine-controller/blob/fce1320fd139f6ee48f762a2e8696f02aa87ed0c/pkg/cloudprovider/provider/azure/provider.go#L142-L147

Comment (embik) [reply pkg/controller/machine/bootstrap.go:255]: (that RHEL version is also fairly old; maybe we should bump it?)

Comment (moadqassem) [reply pkg/controller/machine/bootstrap.go:255]: Should go to 8.5 at least

**Review by embik — APPROVED** [review]

**Comment by embik** [comment]
I can also approve this later on, just giving @moadqassem a chance to take a look as well.

**Review by moadqassem — APPROVED** [review]
/approve 
/lgtm

**Comment by mlavacca** [comment]
> @mlavacca just a note, can you please update the release note to include the two changes in this PR (adding the subscription module configuration and updating the image)? Thanks!

Done :)

**Comment by embik** [comment]
/hold 

seems this is broken, to be in investigated.

**Comment by mlavacca** [comment]
Azure failing tests have been fixed. PTAL @embik @moadqassem

**Review by embik — APPROVED** [review]

**Pull request merged**